scapy/utils.py

```diff
@@ -786,8 +796,12 @@ def read_block_epb(self, block, size):
 
 
 class PcapNgReader(RawPcapNgReader):
-    def __init__(self, filename):
-        RawPcapNgReader.__init__(self, filename)
+
+    alternative = PcapReader
+
+    def __init__(self, filename, fdesc, magic):
+        RawPcapNgReader.__init__(self, filename, fdesc, magic)
+
     def read_packet(self, size=MTU):
         rp = RawPcapNgReader.read_packet(self, size=size)
```

Comment:
What is the expected behavior when self type is PcapReader ? It seems that RawPcapNgReader.read_pack will be called on a legacy pcap file.